spec/govuk_error/configuration_spec.rb

```diff
@@ -96,22 +96,92 @@
+        ClimateControl.modify SENTRY_CURRENT_ENV: "production" do
+          configuration.active_sentry_environments << "production"
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("errors_occurred")
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("error_types.standard_error")
+          configuration.before_send.call(StandardError.new)
+        end
+      end
+    end
+
+    context "when the before_send lambda has been overridden" do
+      before { stub_const("GovukStatsd", double(Module)) }
+      it "increments the appropriate counters" do
+        ClimateControl.modify SENTRY_CURRENT_ENV: "production" do
+          configuration.active_sentry_environments << "production"
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("errors_occurred")
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("error_types.standard_error")
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("hello_world")
+
+          configuration.before_send = lambda do |error_or_event, _hint|
+            GovukStatsd.increment("hello_world")
+            error_or_event
+          end
+
+          configuration.before_send.call(StandardError.new)
+        end
+      end
+    end
+
+    context "when the before_send lambda has been overridden several times, all take effect" do
+      before { stub_const("GovukStatsd", double(Module)) }
+      it "increments the appropriate counters" do
+        ClimateControl.modify SENTRY_CURRENT_ENV: "production" do
+          configuration.active_sentry_environments << "production"
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("errors_occurred")
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("error_types.standard_error")
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("hello_world")
+          expect(GovukStatsd).to receive(:increment).exactly(1).times.with("hello_world_again")
+
+          configuration.before_send = lambda do |error_or_event, _hint|
+            GovukStatsd.increment("hello_world")
+            error_or_event
+          end
+          configuration.before_send = lambda do |error_or_event, _hint|
+            GovukStatsd.increment("hello_world_again")
+            error_or_event
+          end
+
+          configuration.before_send.call(StandardError.new)
+        end
       end
     end
   end
 
-  describe ".should_capture=" do
-    it "Allows apps to add their own `should_capture` callback, that is evaluated alongside the default. If both return `true`, then we should capture, but if either returns `false`, then we shouldn't." do
+  describe ".before_send=" do
+    it "Allows apps to add their own `before_send` callback, that is evaluated alongside the default. If both return `true`, then we should capture, but if either returns `false`, then we shouldn't." do
```

Comment:
This test description needs updating to refer to returning the `error_or_event` or `nil`